hack/build-cross.sh

```diff
@@ -28,6 +28,7 @@ KUBE_BUILD_PLATFORMS=("${KUBE_SERVER_PLATFORMS[@]}")
 kube::golang::build_binaries "${KUBE_SERVER_TARGETS[@]}"
 
 KUBE_BUILD_PLATFORMS=("${KUBE_CLIENT_PLATFORMS[@]}")
+KUBE_STATIC_OVERRIDES=("kubectl")
```

Comment:
Probably. I'm hesitant to force a static build of kubectl just to satisfy a new e2e test for exec+proxy. Maybe we could find a way to produce both static and dynamic builds (e.g. kubectl remains dynamic, add kubectl-static)?

cc @brendandburns @bgrant0607 @lavalamp